include/stateMachineFwk/smTypes.h

```diff
@@ -116,7 +144,25 @@ class SmUtility
     StateIdToStringMap stateToStringMap;
 
 };
-        
+
+class StateMachineContext
+{
+public:
+    Event evt;
+    TempDataBlock* tempDataBlock_mp;
+
+    StateMachineContext():evt(),
+            tempDataBlock_mp(NULL)
+    {
+    }
+
+    void clear()
+    {
+        tempDataBlock_mp = NULL;
+        static Event dummyEvt;
```

Comment:
Should we need this static variable? Trying to understand this code.

Reply:
StateMachineCtxt is used to store the current Event (which has eventId & evtMsg) + selected procedure context for the event. After processing each event, the values stored are cleared, the dummyEvt is just representing the default event used to reset the stateMachineCtxt.evt field (which inturn clears the evtMsg and eventId fields).